pkg/controller/controller.go

```diff
@@ -193,10 +194,16 @@ func NewTypedUnmanaged[request comparable](name string, mgr manager.Manager, opt
 	}
 
 	if options.NewQueue == nil {
-		options.NewQueue = func(controllerName string, rateLimiter workqueue.TypedRateLimiter[request]) workqueue.TypedRateLimitingInterface[request] {
-			return workqueue.NewTypedRateLimitingQueueWithConfig(rateLimiter, workqueue.TypedRateLimitingQueueConfig[request]{
-				Name: controllerName,
-			})
+		if mgr.GetControllerOptions().UsePriorityQueue {
+			options.NewQueue = func(controllerName string, rateLimiter workqueue.TypedRateLimiter[request]) workqueue.TypedRateLimitingInterface[request] {
+				return priorityqueue.New[request](controllerName)
+			}
+		} else {
+			options.NewQueue = func(controllerName string, rateLimiter workqueue.TypedRateLimiter[request]) workqueue.TypedRateLimitingInterface[request] {
+				return workqueue.NewTypedRateLimitingQueueWithConfig(rateLimiter, workqueue.TypedRateLimitingQueueConfig[request]{
+					Name: controllerName,
+				})
+			}
```

Comment:
nit: should we move the `if` condition inside of the `NewQueue` function body?

Reply:
Done